docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -284,6 +284,65 @@ spec:
 
 NOTE: Logstash persistent queues (PQs) and dead letter queues (DLQs) are not currently managed by the Logstash operator, and using them will require you to create and manage your own Volumes and VolumeMounts
 
+[id="{p}-logstash-volumes"]
+=== Defining data volumes for Logstash
+By default, a PersistentVolume called `logstash-data` is created, that maps to `/usr/share/logstash/data`` for persistent storage, typically used for storage from plugins. The `logstash-data` volume claim is, by default, a small (1Gi) volume, using the standard StorageClass of your Kubernetes cluster, but can be overridden by adding a `spec.volumeClaimTemplate` section named `logstash-data`.
+
+[source,yaml,subs="attributes,+macros,callouts"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: logstash
+spec:
+  # some configuration attributes omitted for brevity here
+  volumeClaimTemplates:
+    - metadata:
+        name: logstash-data # Do not change this name unless you set up a volume mount for the data path.
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 2Gi
+----
+
+
+Separate storage (for example for Logstash configurations using persistent queues (PQ) and/or dead letter queues (DLQ) can be added by including an additional `spec.volumeClaimTemplate` along with a corresponding `spec.podTemplate.spec.containers.volumeMount` for each requested volume, as shown in the following example:
+
+[source,yaml,subs="attributes,+macros,callouts"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: logstash
+spec:
+  count: 1
+  version: {version} 
+  podTemplate:
+    spec:
+      containers:
+      - name: logstash
+        volumeMounts:
+          - mountPath: /usr/share/logstash/pq
+            name: pq
+            readOnly: false
+  volumeClaimTemplates:
+    - metadata:
+        name: pq # needs to match the volume mount in the podTemplate section
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 10Gi
+  config: 
+    log.level: info
+    queue.type: persisted
+    path.queue: /usr/share/logstash/pq
+----
+
+NOTE: The current implementation does not allow resizing of volumes even if your chosen storage class would support it. To resize a volume you have to fully delete the Logstash object, delete and recreate or resize the volume and create a new Logstash object. Also note that volume claims will not be deleted when you delete the Logstash object and have to be deleted manually. This behaviour might change in future versions of the ECK operator. 
```

Comment:
We should also mention that this is a breaking change to existing Logstash deployments, as this requires a change to the `StatefulSet` associated with Logstash